### PR TITLE
chore(contrib): Add pre push hook and script

### PR DESCRIPTION
**Pull request by hoangdv2429**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: contrib/githooks/pre-push
Comment (mtsitrin) [inline contrib/githooks/pre-push:1]: @hoangdv2429 why an empty file?

**Review by danwt — CHANGES_REQUESTED** [review]
what are the hooks?

https://pre-commit.com/

what's the use case?

**Review by omritoptix — DISMISSED** [review]

**Comment by danwt** [comment]
@hoangdv2429 ?
https://github.com/dymensionxyz/dymension-rdk/pull/429#pullrequestreview-2082223497

**Comment by hoangdv2429** [comment]
> @hoangdv2429 ? [#429 (review)](https://github.com/dymensionxyz/dymension-rdk/pull/429#pullrequestreview-2082223497)

Sorry, this script was for running golanglint and markdownlint before the push go through. Save time for additional CI run on pushing new commit.

File: contrib/githooks/pre-push
Comment (hoangdv2429) [reply contrib/githooks/pre-push:1]: I updated it

**Review by mtsitrin — APPROVED** [review]

**Comment by artemijspavlovs** [comment]
I would highly prefer if we could use common tooling for enforcing code quality and standards across all of dymension's repos and would urge us to use [pre-commit](https://pre-commit.com/) as pointed out by @danwt .

an example for golangci-lint is available in [roller](https://github.com/dymensionxyz/roller/blob/main/.pre-commit-config.yaml#L11-L14)

**Comment by hoangdv2429** [comment]
> I would highly prefer if we could use common tooling for enforcing code quality and standards across all of dymension's repos and would urge us to use [pre-commit](https://pre-commit.com/) as pointed out by @danwt .
> 
> an example for golangci-lint is available in [roller](https://github.com/dymensionxyz/roller/blob/main/.pre-commit-config.yaml#L11-L14)

As I read the config on roller, commit messages would be enforced to have one of those prefixes. Is this really what we need? I thought we would squash and merge PRs with the PR title as the commit message.

**Comment by artemijspavlovs** [comment]
> As I read the config on roller, commit messages would be enforced to have one of those prefixes. Is this really what we need? I thought we would squash and merge PRs with the PR title as the commit message.

i was referencing just the golangci lint :)

**Comment by hoangdv2429** [comment]
I check again and see that @mtsitrin already added pre-commit to rdk. If we don't need pre-push then I think we can close it.

**Pull request closed without merging**